spring-cloud-modules/spring-cloud-openfeign/src/main/java/com/baeldung/cloud/openfeign/customizederrorhandling/client/ProductClient.java

```diff
@@ -0,0 +1,17 @@
+package com.baeldung.cloud.openfeign.customizederrorhandling.client;
+
+import com.baeldung.cloud.openfeign.customizederrorhandling.config.FeignConfig;
+import com.baeldung.cloud.openfeign.defaulterrorhandling.model.Product;
+
+import org.springframework.cloud.openfeign.FeignClient;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+@FeignClient(name = "product-client-2", url = "http://localhost:8081/product/", configuration = FeignConfig.class)
+public interface ProductClient {
+
+    @RequestMapping(value = "{id}", method = RequestMethod.GET, produces = "application/json")
```

Comment:
Is this correct? Does the `getProduct` produce `application/json`? or does it `consume`?

Reply:
I think it's correct as the Rest Api produces the resource as application/json and earlier article also mentioned the same thing in feign client. https://www.baeldung.com/spring-cloud-openfeign. Still, there is some element of doubt so i have removed that part as it works even without that.